pkg/storage/store.go

```diff
@@ -161,10 +161,21 @@ func (s *store) init() error {
 			return err
 		}
 
-		w, idx, stop, err := s.storeForPeriod(p, chunkClient, f)
+		// getTableRange is used only for initializing boltdb and tsdb index clients.
+		// Computing TableRange for other indexTypes might cause a panic if PeriodConfig.IndexTables.Period is not configured.
```

Comment:
The `PeriodConfig.IndexTables.Period` should always be set. Infact we should do a validation for it.
It would be good to pass `config.TableRange` instead of a function to the factory functions.

Reply:
0 seems like a valid option when configuring [non-periodic](https://github.com/grafana/loki/blob/25db6bcff287dcf6dc60a8250645573c56566b55/pkg/storage/config/schema_config.go#L529) tables for older indexes, CMIIW

Reply:
can we handle that scenario in config.TableRange methods? If `Period` is 0 and table name is same as `Prefix` then just assume table is within the range.